### ISSUE TITLE
Received fatal alert: handshake_failure

### ISSUE THREAD

**Issue by sulaukad** [kubernetes-client/java#917]
I'm trying to get namespace pods from Pod already running in the cluster in the same namespace.
Client configuration:

```
            ApiClient client = ClientBuilder.cluster().build();
            Configuration.setDefaultApiClient(client);
            CoreV1Api api = new CoreV1Api();
            V1PodList v1PodList = api.listNamespacedPod("test-namespace", null, null, null, null, null, null, null, null, null);
            log.debug("pod list {}", v1PodList.toString());
```
But I receive handshake exception

> Caused by: javax.net.ssl.SSLHandshakeException: Received fatal alert: handshake_failure
	at java.base/sun.security.ssl.Alert.createSSLException(Unknown Source) ~[na:na]
	at java.base/sun.security.ssl.Alert.createSSLException(Unknown Source) ~[na:na]
	at java.base/sun.security.ssl.TransportContext.fatal(Unknown Source) ~[na:na]
	at java.base/sun.security.ssl.Alert$AlertConsumer.consume(Unknown Source) ~[na:na]
	at java.base/sun.security.ssl.TransportContext.dispatch(Unknown Source) ~[na:na]
	at java.base/sun.security.ssl.SSLTransport.decode(Unknown Source) ~[na:na]
	at java.base/sun.security.ssl.SSLSocketImpl.decode(Unknown Source) ~[na:na]
	at java.base/sun.security.ssl.SSLSocketImpl.readHandshakeRecord(Unknown Source) ~[na:na]
	at java.base/sun.security.ssl.SSLSocketImpl.startHandshake(Unknown Source) ~[na:na]
	at okhttp3.internal.connection.RealConnection.connectTls(RealConnection.java:336) ~[okhttp-3.14.7.jar:na]
	at okhttp3.internal.connection.RealConnection.establishProtocol(RealConnection.java:300) ~[okhttp-3.14.7.jar:na]
	at okhttp3.internal.connection.RealConnection.connect(RealConnection.java:185) ~[okhttp-3.14.7.jar:na]
	at okhttp3.internal.connection.ExchangeFinder.findConnection(ExchangeFinder.java:224) ~[okhttp-3.14.7.jar:na]

Running pod uses mounted service account certificate and token under `/var/run/secrets/kubernetes.io/serviceaccount`

Also, I'm able to execute the same request using curl and supplying pod's cert and token to it.
I've tried to set `.setVerifyingSsl(false)` but I have the same response.
Java version:

> openjdk 11.0.6 2020-01-14 LTS
OpenJDK Runtime Environment Corretto-11.0.6.10.1 (build 11.0.6+10-LTS)
OpenJDK 64-Bit Server VM Corretto-11.0.6.10.1 (build 11.0.6+10-LTS, mixed mode)

Thanks in advance!

**Comment by markruler** [comment]
Same problem.
I had no problems with "v1.15.0".

java version "1.8.0_241"
Java(TM) SE Runtime Environment (build 1.8.0_241-b07)
Java HotSpot(TM) 64-Bit Server VM (build 25.241-b07, mixed mode)

Mac OS X

kubectl version
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.0", GitCommit:"9e991415386e4cf155a24b1da15becaa390438d8", GitTreeState:"clean", BuildDate:"2020-03-25T14:58:59Z", GoVersion:"go1.13.8", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.0", GitCommit:"9e991415386e4cf155a24b1da15becaa390438d8", GitTreeState:"clean", BuildDate:"2020-03-25T14:50:46Z", GoVersion:"go1.13.8", Compiler:"gc", Platform:"linux/amd64"}

**Comment by rjeberhard** [comment]
Please try with OpenJDK 11.0.7, which came out this week.  There was a bug fix for TLS 1.3 certificate extensions.  I'm now able to connect to K8s 1.16+ clusters where I was not previously with 11.0.6.

**Comment by sulaukad** [comment]
@rjeberhard 
Kubernetes API was missing EC ciphers which do not ship with Correto distribution
I've added a bouncycastle dependency to my build.gradle registered it in java 
>implementation "org.bouncycastle:bcprov-jdk15on"

> Security.insertProviderAt(new BouncyCastleProvider(), 1)
